tests/test_util/test_util_logging.py

```diff
@@ -396,3 +396,16 @@ def test_get_node_location_abspath():
     location = logging.get_node_location(n)
 
     assert location == absolute_filename + ':'
+
+
+@pytest.mark.sphinx(confoverrides={'show_warning_types': True})
+def test_show_warning_types(app, status, warning):
+    logging.setup(app, status, warning)
+    logger = logging.getLogger(__name__)
+    logger.warning('message2')
+    logger.warning('message3', type='test')
+    logger.warning('message4', type='test', subtype='logging')
+
+    assert 'WARNING: message2' in warning.getvalue()
+    assert 'WARNING: message3 [test]' in warning.getvalue()
+    assert 'WARNING: message4 [test.logging]' in warning.getvalue()
```

Comment:
```suggestion
    warnings = strip_escseq(warning.getvalue()).splitlines()
    assert len(warnings) == 3
    assert ws[0].endswith('WARNING: message2')
    assert ws[1].endswith('WARNING: message3 [test]')
    assert ws[2].endswith('WARNING: message4 [test.logging]')
```

I think this should work (I don't remember if there's something before the WARNING; if not, replace the check by `==`).

Reply:
Updated the test (had to deal with the strip color issue 😏)